src/Controller/Component/CacheComponent.php

```diff
@@ -135,7 +135,10 @@ protected function _writeFile(string $content, string $cache) {
 		if (Configure::read('debug') && !is_dir($folder)) {
 			mkdir($folder, 0770, true);
 		}
-
+        $basefolder = $folder . dirname($cache);
+		if (Configure::read('debug') && !is_dir($basefolder)) {
```

Comment:
why is this needed, the true param above also auto creates base folder.

Reply:
I changed the structure to store cached files.

Now the cache directory will have subfolders as show in URL.

For exemple a request to "www.website.com/category/subcategory/my-news-title-goes-here.html" (Cakephp3) will produce a cache file in:
/tmp/cache/views/category/subcategory/my-news-title-goes-here.html

There is another important think, the querystring part (right after ? in URL) will be hashed, for exemple a request to "www.website.com/category/subcategory/my-news-title-goes-here.html?utm_source=somewhere" (Cakephp3) will produce a cache file in:
/tmp/cache/views/category/subcategory/my-news-title-goes-here-fh324j56.html

Working that way there is no problem when you have a huge number of cached files.